go.mod

```diff
@@ -141,23 +148,145 @@ require (
 	golang.org/x/time v0.3.0 // indirect
 	gomodules.xyz/jsonpatch/v2 v2.2.0 // indirect
 	google.golang.org/appengine v1.6.7 // indirect
-	google.golang.org/genproto v0.0.0-20221024183307-1bc688fe9f3e // indirect
-	google.golang.org/grpc v1.50.1 // indirect
+	google.golang.org/genproto v0.0.0-20230306155012-7f2fa6fef1f4 // indirect
+	google.golang.org/grpc v1.53.0 // indirect
 	google.golang.org/protobuf v1.28.1 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/warnings.v0 v0.1.2 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	k8s.io/apiextensions-apiserver v0.26.1 // indirect
-	k8s.io/apiserver v0.26.1 // indirect
-	k8s.io/cli-runtime v0.26.0 // indirect
-	k8s.io/component-base v0.26.1 // indirect
-	k8s.io/klog/v2 v2.80.1 // indirect
+	k8s.io/apiserver v0.26.2 // indirect
+	k8s.io/cli-runtime v0.26.1 // indirect
+	k8s.io/component-base v0.26.2 // indirect
+	k8s.io/klog/v2 v2.90.1 // indirect
 	k8s.io/kube-openapi v0.0.0-20221012153701-172d655c2280 // indirect
-	k8s.io/utils v0.0.0-20221128185143-99ec85e7a448 // indirect
-	oras.land/oras-go v1.2.2 // indirect
+	k8s.io/utils v0.0.0-20230220204549-a5ecb0141aa5 // indirect
+	oras.land/oras-go v1.2.3 // indirect
 	sigs.k8s.io/json v0.0.0-20220713155537-f223a00ba0e2 // indirect
 	sigs.k8s.io/kustomize/api v0.12.1 // indirect
 	sigs.k8s.io/kustomize/kyaml v0.13.9 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
+
+require (
```

Comment:
In that case, should we just do `go install ...tilt@latest` (and friends)?

Because the current go.mod insanity has one "advantage": The versions of these deps is also encoded in the `go.mod` and as such is shared with the actual code.

_But_, I'm also not sure if that really matters - and just installing latest or a fixed version might be completely fine as well. Especially because of the `replace` directives that we needed as well 🙈

Reply:
I think we probably should:
* `go install ...` all the things _in the CI script only_
* Call tilt and friends as executables to avoid the fiddly replace directives
* rely on developers having dependencies installed (document this in a test/README.md perhaps?)

Reply:
Maybe good to know, from `go help install`:
> If the arguments have version suffixes (like @latest or @v1.0.0), "go install"
builds packages in module-aware mode, ignoring the go.mod file in the current
directory or any parent directory, if there is one. This is useful for
installing executables without affecting the dependencies of the main module.

So:

> go install ... all the things in the CI script only

`@latest`? Or at which versions? (I personally think latest would be fine)

> Call tilt and friends as executables to avoid the fiddly replace directives

We do that already, that has nothing to do with the `replace` directives. The `replace` directives were required because of the `go install`.

> rely on developers having dependencies installed 

We do that already as well 😄

Reply:
Regarding the document: It's currently documented in the test header - may not be visible enough though.

However, the test will error out with an error message that should be obvious enough, so maybe we don't need a separate document?

Reply:
Discussed on a quick call, we're clear on this and I've adjusted the PR to use `go install ...@version`, so there's no `go.mod` changes anymore.